pkg/services/ngalert/notifier/external_alertmanager_test.go

```diff
@@ -168,6 +168,61 @@ func TestIntegrationRemoteAlertmanagerSilences(t *testing.T) {
 	require.Equal(t, *silences[1].Status.State, "expired")
 }
 
+func TestIntegrationRemoteAlertmanagerAlerts(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	amURL, ok := os.LookupEnv("AM_URL")
+	if !ok {
+		t.Skip("No Alertmanager URL provided")
+	}
```

Comment:
I thought the same thing and started by not doing this but the CI failed in normal integration tests. I then checked how we handle tests for e.g. [Memcached](https://github.com/grafana/grafana/blob/f9484fcf820338cae658dce4dbc8d89a0ec46037/pkg/infra/remotecache/memcached_storage_integration_test.go#L15-L18), [Redis storage](https://github.com/grafana/grafana/blob/f9484fcf820338cae658dce4dbc8d89a0ec46037/pkg/infra/remotecache/redis_storage_integration_test.go#L18-L21), [Redis cache](https://github.com/grafana/grafana/blob/f9484fcf820338cae658dce4dbc8d89a0ec46037/pkg/services/live/managedstream/cache_redis_test.go#L16-L19), [Postgres](https://github.com/grafana/grafana/blob/f9484fcf820338cae658dce4dbc8d89a0ec46037/pkg/tsdb/postgres/postgres_test.go#L183-L185) and [MySQL](https://github.com/grafana/grafana/blob/f9484fcf820338cae658dce4dbc8d89a0ec46037/pkg/tsdb/mysql/mysql_test.go#L38-L40) which are the other integration tests that use docker containers. All of them use this pattern.

Reply:
Ugh, I don't love this approach but if there's so much precedence I take it it might not be a simple problem. Can you file an issue and attach it in a separate task list to the epic so that can investigate later? There are ways to simple skip these tests if we don't have the right setup in the CI.

Reply:
Issue created: https://github.com/grafana/grafana/issues/76798